ext/mbstring/libmbfl/filters/mbfilter_utf7.c

```diff
@@ -537,8 +537,10 @@ static size_t mb_utf7_to_wchar(unsigned char **in, size_t *in_len, uint32_t *buf
 			if (p == e) {
 				/* It is an error if trailing padding bits are not zeroes or if we were
 				 * expecting the 2nd part of a surrogate pair when Base64 section ends */
-				if ((n3 & 0x3) || surrogate1)
+				if ((n3 & 0x3) || surrogate1) {
 					*out++ = MBFL_BAD_INPUT;
+					surrogate1 = 0;
```

Comment:
I caught on lldb. Could you add test case?

```
mb_convert_encoding("+euJ", "UTF-8", "UTF-7");
```

Reply:
Sure!